### DISCUSSION TITLE
`reportUnusedFunction` with `pytest.fixture`

### DISCUSSION THREAD

**Discussion by belegnar** [microsoft/pyright#8773] [Q&A]
```
from typing import Iterator

import pytest


@pytest.fixture
def _mock() -> Iterator:
    yield


@pytest.fixture
def client(_mock) -> Iterator:
    yield
```

**Comment by erictraut** [comment]
Pyright is working as intended. Pyright has no special knowledge of third-party libraries or the "magic" they perform under the covers. In this case, `pytest` is performing behind-the-scenes registration and injection.

You can disable `reportUnusedFunction` in all of your tests (using an execution environment for your test directory), in individual files (using a `# pyright: reportUnusedFunction=true` comment at the top of the file), or on a line-by-line basis (using a `# pyright: ignore[reportUnusedFunction]` comment).
